src/components/article/AllArticles.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getAllArticles } from '../../actions/articles';
+import './AllArticles.scss';
+
+class AllArticles extends Component {
+  state = {
+    done: false,
+    defaultImg: 'https://tmsnrt.rs/2FNDBT2',
+    defaultText: 'Just a placeholder',
+  };
+
+  async componentDidMount() {
+    const { getAllArticles: getAll } = this.props;
+    await getAll();
+    this.setState({ done: true });
+  }
+
+  handleClick = async () => {
+    const { getAllArticles: getAll } = this.props;
+    await getAll();
+    this.setState({ done: true });
+  };
+
+  renderArticles() {
+    const { allArticles } = this.props;
+    const { done, defaultImg, defaultText } = this.state;
+    if (done) {
+      return allArticles.allArticles.map(article => (
+        <div className="col s12 m6 l6" key={article.id}>
```

Comment:
This should be abstracted out into its own presentational component like, say, `ArticleCard`  since it's not only on this view that we have this form of article card.

Creating a presentation component with receives props for the kind of content to show on the card, would make it easier for every other view (like related articles section on the article page, homepage, profile page) to reuse it.

Reply:
Working on this already.

Reply:
@akhilome I have already refactored it to be a reusable component.